### ISSUE TITLE
Consider using `indexmap` for hash maps

### ISSUE THREAD

**Issue by epatters** [ToposInstitute/CatColab#13]
The Rust package [indexmap](https://lib.rs/crates/indexmap) provides drop-in replacements for `HashSet` and `HashMap` that maintain an ordering on keys. That's almost always what we want, so we should probably just make this our default data structure for hash sets and maps.

**Comment by olynch** [comment]
Question: do we want insertion-order, or do we want to sort them? Arguably, sorting is even *more* deterministic.

**Comment by epatters** [comment]
Good point. I'm not sure which is preferable. One advantage to insertion order is that it will preserve any order coming from the user interface, which might be convenient for debugging.
